heartbeat/ibmcloud-vpc-cr-vip.in

```diff
@@ -0,0 +1,248 @@
+        ext_ip_1 (str, mandatory): First Instance IP.
+        ext_ip_2 (str, mandatory): Secound Instance IP.
+        api_key (str, optional): IBM Cloud API acsess key. Defaults to "".
+
+    Returns:
+        _type_: _description_
+    """
+    ocf.logger.info("stop_action:stoping")
+    res = monitor_action(vpc_url, ext_ip_1, ext_ip_2, api_key)
+    if res == ocf.OCF_NOT_RUNNING:
+        ocf.logger.info("Resource is already stopped")
+    if res == ocf.OCF_SUCCESS:
+        for i in range(0,RETRY_COUNT):
+            time.sleep(SLEEP_TIME)
+            res = monitor_action(vpc_url, ext_ip_1, ext_ip_2, api_key)
+            if res == ocf.OCF_NOT_RUNNING:
+                break
+    return ocf.OCF_SUCCESS
+
+def start_action(
+    vpc_url="",
+    ext_ip_1="",
+    ext_ip_2="",
+    api_key="",
+):
+    """start_action: redirect  the service ip.
+    Args:
+        vpc_url (str, mandatory): vpc_url for teh region .
+        ext_ip_1 (str, mandatory): First Instance IP.
+        ext_ip_2 (str, mandatory): Secound Instance IP.
+        api_key (str, optional): IBM Cloud API acsess key. Defaults to "".
+
+    Change custom route nexthop to point to this endpoint.
+    In case of a cross AZ Active Passive the route adveritise zone will
+    be chaged to teh new acrtive zone
+    """
+    try:
+        ibm_cloud_fail_over.fail_over_cr_vip("SET", vpc_url , ext_ip_1,
+		ext_ip_2, api_key)
+        active_ip = ibm_cloud_fail_over.fail_over_cr_vip("GET", vpc_url ,
+		ext_ip_1, ext_ip_2, api_key)
+        ocf.logger.info(f"start_action: active_ip: {active_ip}")
+    except Exception as e:
+        ocf.logger.error('Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+
+    return ocf.OCF_SUCCESS
+
+def monitor_action(
+    vpc_url="",
+    ext_ip_1="",
+    ext_ip_2="",
+    api_key="",
+):
+    """monitor_action: check if service ip and gateway are responding."""
+
+    ocf.logger.debug("monitor_action:")
+    try:
+        active_ip = ibm_cloud_fail_over.fail_over_cr_vip("GET", vpc_url , ext_ip_1,
+			ext_ip_2, api_key)
+        if ip_address_validator(active_ip):
+            if os_ip(active_ip):
+            	return ocf.OCF_SUCCESS
+        return ocf.OCF_NOT_RUNNING
+    except Exception as e:
+        ocf.logger.error('Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+
+
+def main():
+    """Instantiate the resource agent."""
+
+    agent_description = textwrap.dedent("""\
+        Resource Agent to move a IBM Cloud VIP based on Custom Route
+        from one virtual network interface (VNI) to another.
+        The prerequisites for the use of this resource agent are as follows:
+
+        1. A two-node (VSI or BM) cluster distributed in same Avilability zone or across.
+        2. Enable Instance Metadata enabled on the two nodes
+        3. allow_ip_spoofing enabled on the Virtual network interface
+        3. IBM Cloud API Key or Trused profile:
+
```

Comment:
You should remove the trailing new-lines. They dont make any sense when you run `pcs resource describe <resource>`, in manpages or other representations of the meta-data.